Makefile

```diff
@@ -66,12 +76,41 @@ docker-clean:
 
 ### Tyk Go Plugin ########################################################################
 
+go/src/go.mod:
+	cd ./go/src ; \
+	go mod init tyk-plugin ; \
+	go get -d github.com/TykTechnologies/tyk@`git ls-remote https://github.com/TykTechnologies/tyk.git refs/tags/${TYK_VERSION} | awk '{print $$1;}'` ; \
+	go mod tidy ; \
+	go mod vendor
```

Comment:
By dynamically creating the go.mod and go.sum files as part of the build, we can ensure that the hash version of the tyk library matches the one defined in the Makefile.  And it removes the manual steps for looking up the hash.

Reply:
Love this.  Only concern is if the TYK_VERSION isn't directly specified, it causes this line to fail silently.  I can get passed this, just calling out